catalog/from-k8s/resource_test.go

```diff
@@ -1770,3 +1770,297 @@ func testService(name string) *apiv1.Service {
 		},
 	}
 }
+
+// Test that the proper registrations are generated for a LoadBalancer with a prefix
+func TestServiceResource_lbPrefix(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	client := fake.NewSimpleClientset()
+	syncer := &TestSyncer{}
+
+	// Start the controller
+	closer := controller.TestControllerRun(&ServiceResource{
+		Log:             hclog.Default(),
+		Client:          client,
+		Syncer:          syncer,
+		K8sConsulPrefix: "prefix",
+	})
+	defer closer()
+
+	// Insert an LB service
+	_, err := client.CoreV1().Services(metav1.NamespaceDefault).Create(&apiv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+		},
+
+		Spec: apiv1.ServiceSpec{
+			Type: apiv1.ServiceTypeLoadBalancer,
+		},
+
+		Status: apiv1.ServiceStatus{
+			LoadBalancer: apiv1.LoadBalancerStatus{
+				Ingress: []apiv1.LoadBalancerIngress{
+					apiv1.LoadBalancerIngress{
+						IP: "1.2.3.4",
+					},
+				},
+			},
+		},
+	})
+	require.NoError(err)
+
+	// Wait a bit
+	time.Sleep(500 * time.Millisecond)
+
+	// Verify what we got
+	syncer.Lock()
+	defer syncer.Unlock()
+	actual := syncer.Registrations
+	require.Len(actual, 1)
+	require.Equal("prefixfoo", actual[0].Service.Service)
+	require.Equal("1.2.3.4", actual[0].Service.Address)
+}
+
+//Test externalIP with Prefix
+func TestServiceResource_externalIPPrefix(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	client := fake.NewSimpleClientset()
+	syncer := &TestSyncer{}
+
+	// Start the controller
+	closer := controller.TestControllerRun(&ServiceResource{
+		Log:             hclog.Default(),
+		Client:          client,
+		Syncer:          syncer,
+		K8sConsulPrefix: "prefix",
+	})
+	defer closer()
+
+	// Insert an LB service
+	_, err := client.CoreV1().Services(metav1.NamespaceDefault).Create(&apiv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+		},
+
+		Spec: apiv1.ServiceSpec{
+			Type:        apiv1.ServiceTypeLoadBalancer,
+			ExternalIPs: []string{"a", "b"},
```

Comment:
It would probably be good to have the ExternalIPs resemble what someone might usually find in that field (i.e. "2.3.4.5") rather than an arbitrary string.